### PR TITLE
uboot: Init ubootLibretechAC, using buildUBootMesonGXL

### PR DESCRIPTION
**Pull request by samueldr**
This adds a firmware for the *["La Frite"](https://libre.computer/products/boards/aml-s805x-ac/)* SBC from Libre Computer.

Though, I will say: **the firmware files from upstream are as supported and as good as these**.

This is because the vendor (Libre Computer) uses mainline U-Boot to build their firmware files. Adding support for the *La Frite* SBC was done mainly as an exercise to provide buildUBootMesonGXL to help other users introduce GXL Amlogic boards.

This builds on top of a WIP branch built on top of #101454. The idea being to exercise the new Amlogic U-Boot build infra with a board I can test with.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Additionally:

 - Booted using pyamlusb
 - Booted from eMMC
 - Booted from SPI

### CODE REVIEW

File: pkgs/misc/uboot/amlogic-firmware.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    rm -v LICENSE README.md
+    # Remove unneeded files; we're not re-using the downstream build infra.
+    rm -v */aml_encrypt* */fip_create
+    mkdir -p $out
+    mv -t $out/ *
+  '';
+
+  fixupPhase = ''
+    echo "Fixup phase skipped..."
+  '';
```
Comment (SuperSandro2000) [inline pkgs/misc/uboot/amlogic-firmware.nix:29]:
```suggestion

  dontFixup = true;
```

File: pkgs/misc/uboot/amlogic-firmware.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{ stdenv
+, lib
+, fetchpatch
+, fetchFromGitHub
+}:
+
+stdenv.mkDerivation {
+  pname = "amlogic-firmware";
+  version = "2021-01-29";
```
Comment (SuperSandro2000) [inline pkgs/misc/uboot/amlogic-firmware.nix:9]:
```suggestion
  version = "unstable-2021-01-29";
```

File: pkgs/misc/uboot/amlogic-firmware.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  src = fetchFromGitHub {
+    repo = "amlogic-boot-fip";
+    owner = "LibreELEC";
```
Comment (SuperSandro2000) [inline pkgs/misc/uboot/amlogic-firmware.nix:13]:
```suggestion
    owner = "LibreELEC";
    repo = "amlogic-boot-fip";
```

**Review by lopsided98 — APPROVED** [review]
This boots on my ODROID-C2 and the code looks good to me.

**Comment by samueldr** [comment]
Nice to hear. Note that most of the PR changes are part of #101454, mainly I was interested in your opinion about the amlogic separation of concern, and looks like it's fine :).

I'll rebase and undraft this PR once #101454 is made compatible with the last three commits of this PR, and merged.

**Comment by samueldr** [comment]
Closing in favour of Tow-Boot... and the vendor Platform Firmware. I won't get back to this change set.

Time for some lore: this set of changes, making `buildUBootMesonGXL` and `buildUBootMesonG12A` was influential in the design of Tow-Boot. Working out the family ressemblances and implementing the least repetitive build instructions as I could.

**Pull request closed without merging**